### DISCUSSION TITLE
[BUG]Get-PnPSearchCrawlLog throws access denied although accounts has enough permission

### DISCUSSION THREAD

**Discussion by mikarin20** [pnp/powershell#4386] [General]
### Reporting an Issue or Missing Feature
Please confirm what it is that your reporting

Running Get-PnPSearchCrawlLog but throwing error
Get-PnPSearchCrawlLog: Make sure you are granted access to the crawl log via the SharePoint search admin center at https://-admin.sharepoint.com/_layouts/15/searchadmin/crawllogreadpermission.aspx

my account has connection and permissions on crawllog but still facing issues to access crawl logs. What am i missing?

### Expected behavior 
Please describe what output you expect to see from the PnP PowerShell Cmdlets

see the logs from crawls

### Actual behavior 
Please describe what you see instead. Please provide samples of output or screenshots.

![image](https://github.com/user-attachments/assets/a9f46ea3-8016-4187-aed5-88a38f577d96)

### Steps to reproduce behavior
Please include complete script or code samples in-line or linked from [gists](https://gist.github.com/)
get-pnpsearchcrawllog

### What is the version of the Cmdlet module you are running?
(you can retrieve this by executing ```Get-Module -Name "PnP.PowerShell" -ListAvailable```)

2.12

### Which operating system/environment are you running PnP PowerShell on?
- [x ] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by jackpoz** [comment]
How did you authenticate ?

If I authenticate with -Interactive to the SPO Admin Center with an app registration with delegated permissions SharePoint AllSites.FullControl , I get the log as soon as I add my account to /_layouts/15/searchadmin/crawllogreadpermission.aspx :

![image](https://github.com/user-attachments/assets/bdac18b4-4b9d-4ed6-b517-b78a13289b8c)

**Comment by mikarin20** [comment]
I authenticated with clientid and secret values via app register. It's connecting to spo fine. I'm able to retrieve a lot of information related file size, modified date, names etc but not this one.
I have also full permission like you.

Im not sure what could be missing

**Comment by jackpoz** [comment]
Well, if you connect through clientid and secret, then that's Application permissions, not Delegated permissions, and /_layouts/15/searchadmin/crawllogreadpermission.aspx allows to specify only users, so Delegated permissions.

It might be that this particular API doesn't support application permissions but only delegated permissions, so with -Interactive or -DeviceLogin with an account.

**Reply by mikarin20** [reply]
> Well, if you connect through clientid and secret, then that's Application permissions, not Delegated permissions, and /_layouts/15/searchadmin/crawllogreadpermission.aspx allows to specify only users, so Delegated permissions.
> 
> It might be that this particular API doesn't support application permissions but only delegated permissions, so with -Interactive or -DeviceLogin with an account.

can you show me your full code? Currently im using this one and other options to connect doesnt work for me

Connect-PnPOnline -ClientId $clientID  -url $siteUrl -tenant $tenantID -CertificatePath $certificatepath


![image](https://github.com/user-attachments/assets/61608b99-e1d4-40bb-8237-a35f2939119f)

**Comment by mikarin20** [comment]
upppp
